src/main/java/com/faforever/client/map/generator/MapGeneratorService.java

```diff
@@ -107,62 +115,128 @@ private void deleteGeneratedMaps() {
+    ComparableVersion maxVersion = new ComparableVersion(String.valueOf(clientProperties.getMapGenerator().getMaxSupportedMajorVersion() + 1));
+    if (version.compareTo(maxVersion) >= 0) {
+      CompletableFuture<String> errorFuture = new CompletableFuture<>();
+      errorFuture.completeExceptionally(new IllegalArgumentException("New Version not supported"));
+      return errorFuture;
+    }
+    if (version.compareTo(minVersion) < 0) {
+       CompletableFuture<String> errorFuture = new CompletableFuture<>();
+       errorFuture.completeExceptionally(new IllegalArgumentException("Old Version not supported"));
+       return errorFuture;
+    }
     String generatorExecutableFileName = String.format(GENERATOR_EXECUTABLE_FILENAME, version);
     Path generatorExecutablePath = this.generatorExecutablePath.resolve(generatorExecutableFileName);
 
     CompletableFuture<Void> downloadGeneratorFuture;
     if (!Files.exists(generatorExecutablePath)) {
-      if (!VERSION_PATTERN.matcher(version).matches()) {
-        log.error("Unsupported generator version: {}", version);
-        return CompletableFuture.supplyAsync(() -> {
-          throw new RuntimeException("Unsupported generator version: " + version);
-        });
+      if (!VERSION_PATTERN.matcher(version.toString()).matches()) {
+        log.warn("Unsupported generator version: {}", version);
+        CompletableFuture<String> errorFuture = new CompletableFuture<>();
+        errorFuture.completeExceptionally(new RuntimeException("Unsupported generator version: " + version));
+        return errorFuture;
       }
 
       log.info("Downloading MapGenerator version: {}", version);
       DownloadMapGeneratorTask downloadMapGeneratorTask = applicationContext.getBean(DownloadMapGeneratorTask.class);
-      downloadMapGeneratorTask.setVersion(version);
+      downloadMapGeneratorTask.setVersion(version.toString());
       downloadGeneratorFuture = taskService.submitTask(downloadMapGeneratorTask).getFuture();
     } else {
       log.info("Found MapGenerator version: {}", version);
       downloadGeneratorFuture = CompletableFuture.completedFuture(null);
     }
 
-    String mapFilename = String.format(GENERATED_MAP_NAME, version, seed);
+    String[] seedParts = seedAndOptions.split("_");
+    String seedString = seedParts[0];
+
+    String mapFilename;
+    String seed;
+
+    try {
+      seed = Long.toString(Long.parseLong(seedString));
+    }
+    catch (NumberFormatException nfe) {
+      byte[] seedBytes = Base64.getDecoder().decode(seedString);
+      ByteBuffer seedWrapper = ByteBuffer.wrap(seedBytes);
+      seed = Long.toString(seedWrapper.getLong());
+    }
+
+    // Check if major version 0 which requires numeric seed
+    if (version.compareTo(new ComparableVersion("1")) < 0) {
+      mapFilename = String.format(GENERATED_MAP_NAME, version, seed).replace('/','^');
+    }
+    else {
```

Comment:
```suggestion
    } else {
```

Please make sure that auto-formatting is enabled

Reply:
Done